src/core/ohos/SDL_ohos.c

```diff
@@ -0,0 +1,246 @@
+#include "SDL_internal.h"
+#include <EGL/egl.h>
+#include <EGL/eglplatform.h>
+#include <js_native_api.h>
+#include <js_native_api_types.h>
+#include <node_api.h>
+#include <node_api_types.h>
+
+#ifdef SDL_PLATFORM_OHOS
+
+#include "napi/native_api.h"
+#include "SDL_ohos.h"
+#include <ace/xcomponent/native_interface_xcomponent.h>
+#include "../../video/ohos/SDL_ohosvideo.h"
+#include "SDL3/SDL_mutex.h"
+#include "../../video/ohos/SDL_ohoskeyboard.h"
+
+OHNativeWindow *nativeWindow;
+SDL_Mutex *g_ohosPageMutex = NULL;
+static OH_NativeXComponent_Callback callback;
+static OH_NativeXComponent_MouseEvent_Callback mouseCallback;
+SDL_WindowData windowData;
+SDL_VideoData videoData;
+struct
+{
+    napi_env env;
+    napi_threadsafe_function func;
+    napi_ref interface;
+} napiEnv;
+
+typedef union
+{
+    int i;
+    short s;
+    char c;
+    long long l;
+    float f;
+    double d;
+    const char* str;
+    bool b;
+} napiCallbackArg;
+typedef struct
+{
+    const char* func;
+    napiCallbackArg arg1;
+    napiCallbackArg arg2;
+    napiCallbackArg arg3;
+    napiCallbackArg arg4;
+    napiCallbackArg arg5;
+    napiCallbackArg arg6;
+    napiCallbackArg arg7;
+    napiCallbackArg arg8;
+} napiCallbackData;
+
+static napi_value minus(napi_env env, napi_callback_info info)
+{
+    size_t argc = 2;
+    napi_value args[2] = { NULL };
+
+    napi_get_cb_info(env, info, &argc, args, NULL, NULL);
+
+    napi_valuetype valuetype0;
+    napi_typeof(env, args[0], &valuetype0);
+
+    napi_valuetype valuetype1;
+    napi_typeof(env, args[1], &valuetype1);
+
+    double value0;
+    napi_get_value_double(env, args[0], &value0);
+
+    double value1;
+    napi_get_value_double(env, args[1], &value1);
+
+    napi_value sum;
+    napi_create_double(env, value0 - value1, &sum);
+
+    return sum;
+}
+
+static void sdlJSCallback(napi_env env, napi_value jsCb, void* content, void* data)
+{
+
+}
+
+static napi_value sdlCallbackInit(napi_env env, napi_callback_info info)
+{
+    napiEnv.env = env;
+    size_t argc = 1;
+    napi_value args[1] = { NULL };
+
+    napi_get_cb_info(env, info, &argc, args, NULL, NULL);
+
+    napi_create_reference(env, args[0], 1, &napiEnv.interface);
+
+    napi_value resName = NULL;
+    napi_create_string_utf8(env, "SDLThreadSafe", NAPI_AUTO_LENGTH, &resName);
+    napi_create_threadsafe_function(env, args[0], NULL, resName, 0, 1, NULL, NULL, NULL, sdlJSCallback, &napiEnv.func);
+
+    napi_value result;
+    napi_create_int32(env, 0, &result);
+    return result;
+}
+
+static void OnSurfaceCreatedCB(OH_NativeXComponent *component, void *window)
+{
+    nativeWindow = (OHNativeWindow *)window;
+
+    uint64_t width;
+    uint64_t height;
+    double offsetX;
+    double offsetY;
+    OH_NativeXComponent_GetXComponentSize(component, window, &width, &height);
+    OH_NativeXComponent_GetXComponentOffset(component, window, &offsetX, &offsetY);
+
+    SDL_Log("Native Window: %p", nativeWindow);
```

Comment:
Logging priority level is too high. Either use trace level, or remove this.
There are multiple `SDL_Log`s.